tools/v8-prof/tick-processor.js

```diff
@@ -0,0 +1,67 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//     * Neither the name of Google Inc. nor the names of its
+//       contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+var cp = require('child_process');
+var fs = require('fs');
+var path = require('path');
+
+var toolsPath = path.join(__dirname, '..', '..', 'deps', 'v8', 'tools');
+var scriptFiles = [
+  path.join(__dirname, 'polyfill.js'),
+  path.join(toolsPath, 'splaytree.js'),
+  path.join(toolsPath, 'codemap.js'),
+  path.join(toolsPath, 'csvparser.js'),
+  path.join(toolsPath, 'consarray.js'),
+  path.join(toolsPath, 'csvparser.js'),
+  path.join(toolsPath, 'consarray.js'),
+  path.join(toolsPath, 'profile.js'),
+  path.join(toolsPath, 'profile_view.js'),
+  path.join(toolsPath, 'logreader.js'),
+  path.join(toolsPath, 'tickprocessor.js'),
+  path.join(toolsPath, 'SourceMap.js'),
+  path.join(toolsPath, 'tickprocessor-driver.js')];
+
+var tempScript = path.join(__dirname, 'tick-processor-input-script-$$');
+var cat = cp.spawn('cat', scriptFiles);
+cat.stdout.pipe(fs.createWriteStream(tempScript));
+cat.stdout.on('end', function() {
+  var tickArguments = [tempScript];
+  if (process.platform === 'darwin') {
+    tickArguments.push('--mac', '--nm=' + path.join(toolsPath, 'mac-nm'));
+  } else if (process.platform === 'win32') {
+    tickArguments.push('--windows');
+  }
+  tickArguments.push.apply(tickArguments, [].slice.call(process.argv, 2));
+  console.log(tickArguments);
+  var processTicks = cp.spawn(process.argv[0], tickArguments);
+  processTicks.stdout.on('data', function(chunk) {
+    console.log(chunk.toString());
+  });
```

Comment:
`processTicks.stdout.pipe(process.stdout)`?  `chunk.toString()` doesn't quite work in the presence of partial multi-byte character sequences.